### PR TITLE
Improve template documentation

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

Addresses [template documentation improvements](https://github.com/Concordium/concordium.github.io/issues/630)

I created a READ.me file. This READ.me includes additional information on the content of the templates and which part of the templates should be customized (e.g. the state of the default template or the logic of the receive function of the default template).
Several gifs were also added to show what people can do after generating a smart contract project from these templates (e.g. how to test, compile, and simulate an invoke).
 
This READ.me file will be included in the documentation:
https://github.com/Concordium/concordium.github.io/pull/650

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: templates/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Concordium Smart Contract Templates
+
+This repo helps you to get a new Concordium smart contract project up and running quickly by leveraging the pre-existing templates in this git repository. There are several smart contract templates available in this repository (currently a `default` template and a `cis2-nft` template). For generating the smart contracts from the templates, you need the `cargo-concordium` crate. `Cargo-concordium` can be installed as described in the following guide:
```
Comment (abizjak) [inline templates/README.md:3]:
```suggestion
This repository helps you to get a new Concordium smart contract project up and running quickly by leveraging the pre-existing templates in it. There are several smart contract templates available in this repository (currently a `default` template and a `cis2-nft` template). For generating the smart contracts from the templates, you need the `cargo-concordium` tool. `Cargo-concordium` can be installed as described in the following guide:
```

File: templates/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+[Cargo-concordium setup](https://developer.concordium.software/en/mainnet/smart-contracts/guides/setup-tools.html#setup-tools)
+
+You also need the `cargo-generate` crate. `Cargo-generate` can be installed by running the following command:
```
Comment (abizjak) [inline templates/README.md:7]:
```suggestion
You also need the `cargo-generate` tool. `Cargo-generate` can be installed by running the following command:
```

Comment (abizjak) [reply templates/README.md:7]: Should we also mention at the start that this assumes they have installed `cargo` version at least ...?

File: templates/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+<img src="./gif_recordings/compile.gif" width="700" height="450"/>
+
+To compile your newly generated Concordium smart contract project, run the command:
+
+```
+cargo concordium build -e --out ./my_concordium_project.wasm.v1
+```
+
+This command will compile your new smart contract project and generate a smart contract module file named ``my_concordium_project.wasm.v1``. The ``--out`` option pastes this smart contract module file into your project root folder.
```
Comment (abizjak) [inline templates/README.md:39]: pastes?

Maybe outputs?

File: templates/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+```
+cargo concordium run update --entrypoint receive --module ./my_concordium_project.wasm.v1 --parameter-json throwErrorTrue.json --state-bin state.bin --contract my_concordium_project --context contextUpdate.json
+```
+
+This command simulates an invoke to the ``receive`` function of your smart contract module with the input parameter `throw_error == true`.
+
+## The `cis2-nft` Template
+
+This template generates an NFT token smart contract. The token follows the Concordium [CIS2 token standard](https://proposals.concordium.software/CIS/cis-2.html). To get better familiar with how to interact with a CIS2 token, you can read the
```
Comment (abizjak) [inline templates/README.md:136]:
```suggestion
This template generates an NFT token smart contract. The token follows the Concordium [CIS2 token standard](https://proposals.concordium.software/CIS/cis-2.html). To get more familiar with how to interact with a CIS2 token, you can read the
```

File: templates/README.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+cargo install --locked cargo-generate
+```
+
+Here is an example of using the templates:
+
+## Examples
+
+### Generating a new project
+
+<img src="./gif_recordings/init.gif" width="700" height="450"/>
```
Comment (abizjak) [inline templates/README.md:19]: Could we move all of the images from the top-level into a directory like `resources`?

**Review by abizjak — APPROVED** [review]
Nice addition.

File: templates/README.md (hunk 2 above)
Comment (DOBEN) [reply templates/README.md]: I added the following version info (cargo-generate might work with lower versions as well but I haven't tested it. I recommend the up-to-date version for cargo-generate):

- For generating the smart contracts from the templates, you need the `cargo-concordium` tool **version 2.2.0 or greater**.

- You also need the `cargo-generate` tool **(recommended version 0.16.0 or greater)**.

**Review by abizjak — APPROVED** [review]

File: templates/README.md (hunk 2 above)
Comment (abizjak) [reply templates/README.md]: Great. looks good.

**Review by mh-concordium — APPROVED** [review]
Looks great, I don't have any comments. I think this really adds to understanding of what templates are and how one can try to use them.

**Pull request merged**